Brokerages/WebSocketWrapper.cs

```diff
@@ -63,17 +55,110 @@ public void Initialize(string url)
         /// <param name="data"></param>
         public void Send(string data)
         {
-            _wrapped.Send(data);
+            try
+            {
+                if (!IsOpen)
+                {
+                    Log.Trace($"WebSocketWrapper.Send(): Connection not open (IsOpen:{IsOpen}, State:{_wrapped.State}): attempting connection", true);
+
+                    Connect();
+                }
+
+                var sendBuffer = new ArraySegment<byte>(Encoding.UTF8.GetBytes(data));
+
+                _wrapped.SendAsync(sendBuffer, WebSocketMessageType.Text, true, _cancellationTokenSource.Token).SynchronouslyAwaitTask();
+            }
+            catch (Exception exception)
+            {
+                OnError(new WebSocketError(exception.Message, exception));
+            }
         }
 
         /// <summary>
         /// Wraps Connect method
         /// </summary>
         public void Connect()
         {
-            if (!IsOpen)
+            if (IsOpen)
+            {
+                return;
+            }
+
+            try
+            {
+                _wrapped.ConnectAsync(new Uri(_url), _cancellationTokenSource.Token).SynchronouslyAwaitTask();
+
+                if (IsOpen)
+                {
+                    OnOpen();
+
+                    new TaskFactory().StartNew(
+                        async () =>
+                        {
+                            try
+                            {
+                                await StartReceiving();
+                            }
+                            catch (Exception exception)
+                            {
+                                OnError(new WebSocketError(exception.Message, exception));
+                            }
+
+                            if (_wrapped.State == WebSocketState.Aborted)
```

Comment:
`StartReceiving()->ReceiveAsync()` could throw if for some reason we are not connected, in this case, we will stop receiving until someone calls `Send()` or `Open()`, should we try to reconnect ourselves instead?

Reply:
or even worse, the server can forcefully close the socket w/out sending a close message and you won't know until you call `ReceiveAsync()`

Reply:
_also_ -- reco using `.ConfigureAwait(false)` since some of these components may be used by applications with a message/event loop (anything w/ a UI)